src/execution_plan/ops/op_conditional_traverse.c

```diff
@@ -105,18 +105,20 @@ OpBase *NewCondTraverseOp(const ExecutionPlan *plan, Graph *g, AlgebraicExpressi
 }
 
 static OpResult CondTraverseInit(OpBase *opBase) {
-	CondTraverse *op = (CondTraverse *)opBase;
-	AST *ast = ExecutionPlan_GetAST(opBase->plan);
-	op->recordsCap = TraverseRecordCap(ast);
-	op->records = rm_calloc(op->recordsCap, sizeof(Record));
+	OpCondTraverse *op = (OpCondTraverse *)opBase;
+	// Create 'records' with this Init function as 'record_cap'
+	// might be set during optimization time.
+	// If no cap is specified, use 16 as the default value.
+	if(op->record_cap == UNLIMITED) op->record_cap = 16;
```

Comment:
```suggestion
	if(op->record_cap > 16) op->record_cap = 16;
```
You are setting the `record_cap` with `applyLimit`. Use the `16` optimization for bigger limits.

Reply:
Good catch!